wait-for-couchbase.sh

```diff
@@ -1,29 +1,106 @@
 #!/bin/bash
 # wait-for-couchbase.sh
 
-# set -e
+set -e
 
 CB_HOST="${CB_HOST:-db}"
 CB_USER="${CB_USER:-Administrator}"
 CB_PSWD="${CB_PSWD:-password}"
 
-URL=http://${CB_HOST}:8091/pools/default/buckets/travel-sample/scopes/
 
-echo "wait-for-couchbase: checking $URL"
+#### Utility Functions ####
+# (see bottom of file for the calling script)
 
-out=/tmp/pools.output
-while true; do
-  status=$(curl -s -w "%{http_code}" -o $out -u "${CB_USER}:${CB_PSWD}" $URL)
-  if [ "x$status" = "x200" ]; then
-    echo "wait-for-couchbase: polling for expected scopes"
-    hasScopes=$(jq '.scopes | map(.name) | contains(["inventory", "tenant_agent_00", "tenant_agent_01"])' < $out)
-    if [ "x$hasScopes" = "xtrue" ]; then
-      break
+log() {
+  echo "wait-for-couchbase: $@" | cut -c -${COLUMNS:-80}
+}
+
+wait-for-one() {
+  ATTEMPTS=$1
+  URL=$2
+  QUERY=$3
+  EXPECTED=true
+  OUT=$(mktemp)
+
+  for attempt in $(seq 1 $ATTEMPTS )
+  do
+    status=$(curl -s -w "%{http_code}" -o $OUT -u "${CB_USER}:${CB_PSWD}" $URL)
+    if [ $attempt -eq 1 ]; then
+      log "polling for '$QUERY'"
+      if [ $DEBUG ]; then jq . $OUT; fi
+    elif (( $attempt % 5 == 0 )); then
+      log "..."
+    fi
+    if [ "x$status" == "x200" ]; then
+      result=$(jq "$QUERY" < $OUT)
+      if [ "x$result" == "x$EXPECTED" ]; then
+        return # success
+      fi
+      if [ $attempt -eq 1 ]; then
+        log "value is currently:"
+        jq . <<< "$result"
+      fi
+    fi
+    sleep 2
+  done
+  return 1 # failure
+}
+
+wait-for() {
+  ATTEMPTS=$1
+  URL="http://${CB_HOST}${2}"
+  shift; shift;
+
+  log "checking $URL"
+
+  for QUERY in "$@"
+  do
+    wait-for-one $ATTEMPTS $URL "$QUERY" || ( log "Failure"; exit 1001 )
+  done
+  return # success
+}
+
+function createHotelsIndex() {
+    log "Creating hotels-index..."
+    http_code=$(curl -o hotel-index.out -w '%{http_code}' -s -u ${CB_USER}:${CB_PSWD} -X PUT \
+        http://${CB_HOST}:8094/api/index/hotels-index \
+        -H 'cache-control: no-cache' \
+        -H 'content-type: application/json' \
+        -d @fts-hotels-index.json)
+    if [[ $http_code -ne 200 ]]; then
+        log Hotel index creation failed
+        cat hotel-index.out
+        exit 1000
     fi
-  fi
-  sleep 2
-done
-echo "wait-for-couchbase: travel-sample scopes ready"
+}
+
+##### Script starts here #####
+TIMES=150
```

Comment:
Not sure what `TIMES` represents, is this in seconds?

Reply:
this is the number of times it attempts to poll, suggest a better name?

Reply:
`ATTEMPTS`? Think you used it elsewhere, seems to make sense 👍